allennlp/data/dataset_readers/coreference_resolution/conll.py

```diff
@@ -165,6 +187,23 @@ def text_to_instance(
             for start, end in enumerate_spans(
                 sentence, offset=sentence_offset, max_span_width=self._max_span_width
             ):
+                if self._wordpiece_modeling_tokenizer is not None:
+                    start = offsets[start][0]
+                    end = offsets[end][1]
+
+                    # `enumerate_spans` uses word-level width limit; we need to be stricter here
```

Comment:
This comment confuses me a bit. IIUC, the meaning is that the next statement ensures `_max_span_width` operates on word pieces instead of words. I might say, "`enumerate_spans` uses a word-level width limit; here we apply it to word pieces." The phrasing "we need to be stricter here" sounds almost like a todo comment.

As an aside, did you use the same `_max_span_width` in your experiments with and without the label distribution logic? It seems like that might affect the perf. Actually, dropping this check might allow for a fairer comparison. That said, I don't have much domain knowledge w.r.t. coref, so please let me know if this is a bad/irrelevant idea for some reason.

Reply:
We can't drop it here because we use a span width embedding that has only `_max_span_width` entries, and since we are doing wordpiece modeling, the span width embedding operates on wordpiece lengths. So a check here is necessary or else we wouldn't know how many entries that would be.

And yes I agree that controlling this across word vs. wordpiece modeling experiments is not entirely trivial, but that's a config choice and shouldn't be affected on the code.

Reply:
Thanks. Could you please update the comment in the code with that explanation?